.github/scripts/curate_autogenerated_schema.py

```diff
@@ -0,0 +1,1999 @@
+        )
+
+    # Allow for patternProperties in the envDef array by updating the envDef to be a oneOf array
+    schema_dict["definitions"]["EnvVarRequirement"]["properties"]["envDef"] = {
+        "description": schema_dict["definitions"]["EnvVarRequirement"]["properties"]["envDef"].get("description", ""),
+        "oneOf": [
+            {
+                "items": schema_dict["definitions"]["EnvVarRequirement"]["properties"]["envDef"].get("items", {}),
+                "type": schema_dict["definitions"]["EnvVarRequirement"]["properties"]["envDef"].get("type", {})
+            },
+            {
+                "patternProperties": {
+                    "^[_a-zA-Z][a-zA-Z0-9_-]*$": {
+                        "type": "string"
+                    }
+                }
+            }
+        ]
+    }
+
+    # Return schema dict
+    return schema_dict
+
+
+def add_cwl_metadata_to_schema(schema_dict: Dict) -> Dict:
+    """
+    Add in the CWL metadata to the schema
+    Derived from https://github.com/common-workflow-language/cwl-v1.2/blob/76bdf9b55e2378432e0e6380ccedebb4a94ce483/json-schema/cwl.yaml#L2231-L2241
+    :param schema_dict:
+    :return:
+    """
+
+    # Always do a deepcopy on the input
+    schema_dict = deepcopy(schema_dict)
+
+    # Assert defintions
+    assert_definitions_key(schema_dict)
+
+    # Add in the CWL metadata to the definitions
+    schema_dict["definitions"].update(
+        {
+            "CWLDocumentMetadata": {
+                "description": "Metadata for a CWL document",
+                "type": "object",
+                "properties": {
+                    "$namespaces": {
+                        "description": "The namespaces used in the document",
+                        "type": "object",
+                        "patternProperties": {
+                            "^[_a-zA-Z][a-zA-Z0-9_-]*$": {
+                                "type": "string"
+                            }
+                        }
+                    },
+                    "$schemas": {
+                        "description": "The schemas used in the document",
+                        "type": "array",
+                        "items": {
+                            "type": "string"
+                        }
+                    }
+                },
+                "patternProperties": {
+                    "^s:.*$": {
+                        "type": "object"
+                    },
+                    # Or the full version
+                    "^https://schema.org/.*$": {
+                        "type": "object"
+                    },
+                    # dct terms should also be permitted
+                    # see https://github.com/common-workflow-language/cwl-v1.2/blob/main/tests/metadata.cwl
+                    "^dct:.*$": {
+                        "type": "object"
+                    },
+                    "^http://purl.org/dc/terms/.*$": {
+                        "type": "object"
+                    }
```

Comment:
it is valid CWL to have additional prefixed/namespaced  elements at any level, of any namespace; not just schema.org, DCT, etc..

Reply:
I'm reluctant to just put in a `"^.*$"` for pattern properties here, it means spelling mistakes or syntax errors aren't necessarily picked up by the schema. `CWLVersion`, `cwlVersion`, `cwl_version` would all be valid entries

Reply:
I agree that accepting any string anywhere is bad for validation. This is where having a full CWL language server like Benten can go beyond what JSON Schema can do. If you can allow keys that have a colon (`:`) in them, then that should suffice.

Reply:
Resolved in 554c0f20262261bf72cff2278ffe319c7b67a91f / 986742842aea97c1b21a30caafda62f19bcbe36a